ui/widgets/button.js

```diff
@@ -88,6 +89,10 @@ $.widget( "ui.button", {
 
 		this.hasTitle = !!this.element.attr( "title" );
 
+		if ( this.options.textDir ) {
+			this.options.label = this._applyTextDir( this.options.label );
```

Comment:
As a matter of fact inserted UCC constitutes just text node in HTML markup and doesn't break it, while having desirable impact.
However I made changes accounting to possibility of HTML content of button widget and now the element 'direction' style gets changed instead of insertion of directional UCC.